Remotion/ObjectBinding/Web.ClientScript/Scripts/BocReferenceValueBase.ts

```diff
@@ -134,73 +134,83 @@ class BocReferenceValueBase //TODO RM-7715 - Make the TypeScript classes BocRefe
+        newCommand.setAttribute(oldCommandAttributes[i].nodeName, value);
     }
 
     for (var property in tempCommandInfo)
     {
       const value = tempCommandInfo[property];
       if (value == null)
-        newCommand.removeAttr(property);
+        newCommand.removeAttribute(property);
       else
-        newCommand.attr(property, value);
+        newCommand.setAttribute(property, value);
     }
 
-    newCommand.removeClass('hasIcon');
+    newCommand.classList.remove('hasIcon');
 
     return newCommand;
   };
 
-  public static CreateEmptyIcon (oldIcon: JQuery, title: Nullable<string>): JQuery
+  public static CreateEmptyIcon (oldIcon: HTMLImageElement, title: Nullable<string>): HTMLImageElement
   {
     ArgumentUtility.CheckNotNull('oldIcon', oldIcon);
     ArgumentUtility.CheckTypeIsString('title', title);
   
-    var newIcon = oldIcon.clone();
-    newIcon.attr({ src: BocReferenceValueBase._nullIconUrl, alt: '' });
-    newIcon.removeAttr('title');
+    var newIcon = oldIcon.cloneNode(true) as HTMLImageElement;
+    
+    newIcon.setAttribute('src', BocReferenceValueBase._nullIconUrl!);
+
+    newIcon.setAttribute('alt', '');
+    newIcon.removeAttribute('title');
     if (!StringUtility.IsNullOrEmpty(title))
-      newIcon.attr({ title: title });
-    newIcon.css({ width: oldIcon.width(), height: oldIcon.height() });
+      newIcon.setAttribute('title', title);
+
+    newIcon.style.width = LayoutUtility.GetWidth(oldIcon) + 'px';
+    newIcon.style.height = LayoutUtility.GetHeight(oldIcon) + 'px';
   
     return newIcon;
   };
 
-  public static UpdateIconFromWebService (command: JQuery, icon: JQuery, iconInformation: BocReferenceValueBase_IconInformation): void
+  public static UpdateIconFromWebService (command: HTMLElement, icon: HTMLImageElement, iconInformation: Nullable<BocReferenceValueBase_IconInformation>): void
   {
     ArgumentUtility.CheckNotNull('icon', icon);
     ArgumentUtility.CheckTypeIsObject('iconInformation', iconInformation);
   
     if (iconInformation == null)
       return;
   
-    icon.attr({ src: iconInformation.Url });
+    icon.setAttribute('src', iconInformation.Url);
   
-    icon.attr({ alt: '' });
+    icon.setAttribute('alt', '');
     if (!StringUtility.IsNullOrEmpty(iconInformation.AlternateText))
-      icon.attr({ alt: iconInformation.AlternateText });
+      icon.setAttribute('alt', iconInformation.AlternateText);
   
-    if (!StringUtility.IsNullOrEmpty(iconInformation.ToolTip) && StringUtility.IsNullOrEmpty(icon.prop('title')))
-      icon.attr({ title: iconInformation.ToolTip });
+    if (!StringUtility.IsNullOrEmpty(iconInformation.ToolTip) && StringUtility.IsNullOrEmpty(icon.title))
+      icon.setAttribute('title', iconInformation.ToolTip);
   
-    icon.css({ width: iconInformation.Width, height: iconInformation.Height });
+    icon.style.width = TypeUtility.IsString(iconInformation.Width) ? iconInformation.Width : iconInformation.Width + 'px';
```

Comment:
Hmm... In C#, on the IconProxy we're using Unit.ToString() to get the string value that will be serialized by the ASMX web service. In JS, the IconInformation type is typed as number. I think the number-thing's an oversight from the previous PR (similiar to IconUrl being nullable/not-nullable which already got fixed in this PR). Do you want to clean this up right now or make a separate ticket?

Reply:
If it is always a string we can remove the special handling here and adjust the type.

Reply:
Yep, it's always a string with a unit at the end.